opentelemetry-jaeger/src/exporter/agent.rs

```diff
@@ -120,30 +120,74 @@ impl<R: JaegerTraceRuntime> AgentAsyncClientUdp<R> {
             conn,
             buffer_client: BufferClient { buffer, client },
             max_packet_size,
+            auto_split,
         })
     }
 
     /// Emit standard Jaeger batch
     pub(crate) async fn emit_batch(&mut self, batch: jaeger::Batch) -> thrift::Result<()> {
-        // Write payload to buffer
-        self.buffer_client.client.emit_batch(batch)?;
-        let payload = self.buffer_client.buffer.take_bytes();
-
-        if payload.len() > self.max_packet_size {
-            return Err(thrift::ProtocolError::new(
-                thrift::ProtocolErrorKind::SizeLimit,
-                format!(
-                    "jaeger exporter payload size of {} bytes over max UDP packet size of {} bytes. Try setting a smaller batch size.",
-                    payload.len(),
-                    self.max_packet_size,
-                ),
-            )
-            .into());
+        let mut buffers = vec![];
+        split_batch(
+            &mut self.buffer_client,
+            batch,
+            self.auto_split,
+            self.max_packet_size,
+            &mut buffers,
+        )?;
+
+        for payload in buffers {
+            self.runtime.write_to_socket(&self.conn, payload).await?;
         }
 
-        // Write async to socket, reading from buffer
-        self.runtime.write_to_socket(&self.conn, payload).await?;
-
         Ok(())
     }
 }
+
+fn split_batch(
+    client: &mut BufferClient,
+    mut batch: jaeger::Batch,
+    auto_split: bool,
```

Comment:
It feels wired to call the `split_batch` function if you don't want to split at all(auto split is false). It may be better to check it before calling the `split_batch` function. If the auto split is `false`. We could just error out if the payload is too large.

Reply:
There might be more dump code out of `split_batch`, that's why I combine auto split check in this function.

Reply:
That's true. But if the users set the `auto_split` to `false`, we should avoid batch clone in L153 to reduce the overhead

Reply:
> Overall looks like a good solution. Thanks a lot. Could you add some unit tests around it?

To be honest, I don't know how to test this part code, is there any example for me?

Reply:
> we should avoid batch clone in L153 to reduce the overhead

I've updated some code, reduced overhead if `auto_split` is false.

Reply:
> To be honest, I don't know how to test this part code, is there any example for me?

Hmm, I think you are right, it's tricky to simulate the `emit_batch` function here. I guess we could merge it without test and revisit it when we figure out how to test the jaeger exporter.